fix: Flush pool as late as possible during schema cache reloading (#4645)
PostgREST/postgrest: CHANGELOG.md

- Add config `db-timezone-enabled` for optional querying of timezones by @taimoorzaeem in #4751
- Log when the pool is released during schema cache reload on `log-level=debug` by @mkleczek in #4668

### Fixed

> - Pool should be flushed not before but after schema cache reloading by @mkleczek in #4645

Reviewer: More user facing:

```suggestion
- Fix unnecessary connection pool flushes during schema cache reloading by @mkleczek in #4645
```
